spring-cloud-gcp-starters/spring-cloud-gcp-starter-core/src/main/java/org/springframework/cloud/gcp/core/GCPProperties.java

```diff
@@ -27,7 +28,7 @@
 
 	private String projectId;
 
-	private String jsonKeyFile;
+	private Resource credentialKeyFile;
```

Comment:
Yes, and that's not an issue. Think about Datasources again, simple plain old JDBC datasources.

When you import spring-data-jdbc it will look for a property called `spring.datasource.username` and `spring.datasource.password`, at the same time a full blown DataSource object is created in the context. 

Now, the users can if they want to access the username/password (analogy to the resourceFile) and create their own datasource (analogy to GoogleCredentials)

And as a matter of fact, this is how bypassing autoconfiguration works, someone may decide to create the Credentials in a different manner.

So having the constructor primitive for your credentials and a credentials object is not a problem at all.